synapse/storage/databases/main/devices.py

```diff
@@ -1441,6 +1441,13 @@ def __init__(
             self._remove_duplicate_outbound_pokes,
         )
 
+        self.db_pool.updates.register_background_index_update(
+            "device_lists_changes_in_room_by_room_index",
+            index_name="device_lists_changes_in_room_by_room_idx",
+            table="device_lists_changes_in_room",
+            columns=["room_id", "stream_id"],
```

Comment:
It looks like this made a big difference to database disk reads when it was deployed to `matrix.org`😮

https://grafana.matrix.org/d/000000009/postgres?orgId=1&refresh=1m&var-data_source=Prometheus&var-database=matrix&var-bucket_size=%24__auto_interval_bucket_size&var-instance=manticore.matrix.org%3A9113&from=1668993778198&to=1669252978198&viewPanel=51

![](https://user-images.githubusercontent.com/558581/203673432-73b2649c-d053-4d18-85d3-22e8f6d401c7.png)